network/codec/errors.go

```diff
@@ -5,6 +5,9 @@ import (
 	"fmt"
 )
 
+// ErrInvalidEncoding is returned when attempting to decoded a message with an invalid encoding.
+var ErrInvalidEncoding = errors.New("invalid encoding")
```

Comment:
Can we make this a little more descriptive or is it generic on purpose ?

Reply:
addressed in 9c0fb6f